File: python-sdk/src/formanceapi/models/shared/dummypayconfig.py
import dataclasses
from dataclasses_json import dataclass_json
from formanceapi import utils
from typing import Optional


@dataclass_json
@dataclasses.dataclass
class DummyPayConfig:
    directory: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('directory') }})
    file_generation_period: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('fileGenerationPeriod') }})
    file_polling_period: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('filePollingPeriod') }})
    

File: ts-sdk/src/sdk/models/shared/dummypayconfig.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";


export class DummyPayConfig extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=directory" })
  directory: string;

  @SpeakeasyMetadata({ data: "json, name=fileGenerationPeriod" })
  fileGenerationPeriod?: string;

  @SpeakeasyMetadata({ data: "json, name=filePollingPeriod" })
  filePollingPeriod?: string;
}